modules/sfp_threatjammer.py

```diff
@@ -0,0 +1,205 @@
+                "https://threatjammer.com/docs/how-threat-jammer-works",
+                "https://threatjammer.com/docs/introduction-threat-jammer-user-api",
+                "https://threatjammer.com/docs/introduction-threat-jammer-report-api",
+                "https://threatjammer.com/tutorials/how-to-configure-fail2ban-in-ubuntu",
+                "https://threatjammer.com/tutorials/how-to-configure-cowrie-honeypot"
+            ],
+            'apiKeyInstructions': [
+                "https://threatjammer.com/docs/threat-jammer-api-keys",
+                "Register a new account with an email",
+                "Navigate to https://threatjammer.com/keys",
+                "The API Keys are listed under 'API Key' column"
+            ],
+            'favIcon': "https://threatjammer.com/favicon.ico",
+            'logo': "https://threatjammer.com/threatjammer-risk-score.png",
+            'description': "Threat Jammer is a service to access high-quality threat intelligence"
+            " data from a variety of sources, and integrate it into their applications with the"
+            " sole purpose of detecting and blocking malicious activity."
+        }
+    }
+
+    opts = {
+        'api_key': '',
+        "risk_score_min": 35,
+        'checkaffiliates': True,
+    }
+
+    optdescs = {
+        'api_key': "ThreatJammer.com API key.",
+        "risk_score_min": "Minimum Threat Jammer risk score.",
+        'checkaffiliates': "Apply checks to affiliates?",
+    }
+
+    results = None
+
+    def setup(self, sfc, userOpts=dict()):
+        self.sf = sfc
+        self.results = self.tempStorage()
+
+        for opt in list(userOpts.keys()):
+            self.opts[opt] = userOpts[opt]
+
+    def watchedEvents(self):
+        return [
+            "IP_ADDRESS",
+            "IPV6_ADDRESS",
+            "AFFILIATE_IPADDR",
+            "AFFILIATE_IPV6_ADDRESS",
+        ]
+
+    def producedEvents(self):
+        return [
+            "BLACKLISTED_IPADDR",
+            "BLACKLISTED_AFFILIATE_IPADDR",
+            "MALICIOUS_IPADDR",
+            "MALICIOUS_AFFILIATE_IPADDR",
+        ]
+
+    def queryIp(self, ip):
+        """Query API for an IPv4 or IPv6 address.
+
+        Args:
+            ip (str): IP address
+
+        Returns:
+            str: API response as JSON
+        """
+
+        headers = {
+            'Authorization': f"Bearer {self.opts['api_key']}",
+            'Accept': 'application/json',
+        }
+
+        res = self.sf.fetchUrl(
+            f"https://dublin.api.threatjammer.com/v1/assess/ip/{ip}",
+            timeout=self.opts['_fetchtimeout'],
+            useragent=self.opts['_useragent'],
+            headers=headers
+        )
+
+        time.sleep(1)
+
+        if res['code'] == '429':
+            self.error("You are being rate-limited by ThreatJammer.com")
+            self.errorState = True
+            return None
+
+        if res['code'] != "200":
```

Comment:
Does the server reply with an error when the provided API key is invalid (401 / 403) or quota ? If so, this function should handle the appropriate code explicitly and set `errorState` with an appropriate error, rather than presume all non-200 codes are fatal.

Reply:
Added better error handling